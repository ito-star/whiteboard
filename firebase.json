{
  "database": {
    "predeploy": [
      "node \"$PROJECT_DIR/scripts/ensure-master-branch.js\""
    ],
    "rules": "database.rules.json"
  },
  "firestore": {
    "predeploy": [
      "node \"$PROJECT_DIR/scripts/ensure-master-branch.js\""
    ],
    "rules": "firestore.rules",
    "indexes": "firestore.indexes.json"
  },
  "functions": {
    "predeploy": [
      "node \"$PROJECT_DIR/scripts/ensure-master-branch.js\""
    ],
    "source": "functions"
  },
  "hosting": {
    "predeploy": [
      "node \"$PROJECT_DIR/scripts/ensure-master-branch.js\"",
      "node \"$PROJECT_DIR/scripts/build-frontend.js\""
    ],
    "public": "build",
    "ignore": [
      "firebase.json",
      "**/.*",
      "**/node_modules/**",
      "**/*.log"
    ],
    "rewrites": [
      {
        "source": "/oidc/interaction/*",
        "destination": "/index.html"
      },
      {
        "regex": "/oidc(?:/.*)?",
        "function": "oidc"
      },
      {
        "source": "/api-docs",
        "function": "api_docs"
      },
      {
        "source": "/api-docs/**",
        "function": "api_docs"
      },
      {
        "source": "/api-specs/**",
        "function": "api_specs"
      },
      {
        "source": "**",
        "destination": "/index.html"
      }
    ],
    "headers": [
      {
        "source": "**",
        "headers": [
          {
            "key": "Referrer-Policy",
            "value": "strict-origin-when-cross-origin"
          },
          {
            "key": "X-Content-Type-Options",
            "value": "nosniff"
          },
          {
            "key": "X-Frame-Options",
            "value": "SAMEORIGIN"
          },
          {
            "key": "X-XSS-Protection",
            "value": "1; mode=block"
          },
          {
            "key": "Strict-Transport-Security",
            "value": "max-age=31536000; includeSubDomains"
          }
        ]
      },
      {
        "source": "/oidc/**",
        "headers": [
          {
            "key": "Pragma",
            "value": "no-cache"
          },
          {
            "key": "Cache-Control",
            "value": "private, no-cache, no-store, must-revalidate"
          }
        ]
      }
    ]
  },
  "storage": {
    "predeploy": [
      "node \"$PROJECT_DIR/scripts/ensure-master-branch.js\""
    ],
    "rules": "storage.rules"
  },
  "emulators": {
    "database": {
      "host": "localhost",
      "port": 9002
    },
    "functions": {
      "port": 5001
    },
    "firestore": {
      "port": 8080
    },
    "pubsub": {
      "port": 8085
    },
    "ui": {
      "enabled": true,
      "port": 4000
    }
  }
}
