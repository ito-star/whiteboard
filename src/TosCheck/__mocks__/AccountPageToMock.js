import React from "react";

const AccountPageToMock = () => <div>A AccountPage mock passed!</div>;
export default AccountPageToMock;
