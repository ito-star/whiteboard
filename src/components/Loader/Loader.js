import React from "react";
import ReactLoading from "react-loading";
import styled, { css } from "styled-components";

const LoadingWrapper = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  background: rgba(255, 255, 255, 0.15);
  flex-direction: column;

  ${(props) =>
    props.isFullScreen &&
    css`
      position: fixed;
      z-index: 9999;
    `}
`;

const Loader = ({
  id = "app-loader",
  type = "bars",
  color = "#2c387e",
  width,
  height,
  delay,
  hasOverlay = true,
  isFullScreen = false,
  className,
  children,
}) => {
  const Wrapper = hasOverlay ? LoadingWrapper : "div";

  const wrapperProps = {
    className,
  };

  if (hasOverlay) {
    wrapperProps.isFullScreen = isFullScreen;
  }

  return (
    <Wrapper {...wrapperProps}>
      {children}
      <ReactLoading
        id={id}
        className="react-loading"
        type={type}
        color={color}
        width={width}
        height={height}
        delay={delay}
      />
    </Wrapper>
  );
};

export default Loader;
