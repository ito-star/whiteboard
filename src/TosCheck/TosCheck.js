import React, { useEffect } from "react";
import PropTypes from "prop-types";
import firebase from "firebase/compat/app";
import "firebase/compat/database";
import Alert from "react-bootstrap/Alert";
import SwipeableViews from "react-swipeable-views";
import { makeStyles, useTheme } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import TabContext from "@material-ui/lab/TabContext";
import TabPanel from "@material-ui/lab/TabPanel";
import TabList from "@material-ui/lab/TabList";
import Tab from "@material-ui/core/Tab";
import _forEach from "lodash/forEach";

import { initFirebase } from "../utils";
import useUser from "../auth/useUser";
import SimpleNavBar from "../SimpleNavBar";
// import useSubmit from "./useSubmit";
// eslint-disable-next-line import/no-webpack-loader-syntax
import AcceptableUsePolicy, {
  frontMatter as aupFrontMatter,
  // eslint-disable-next-line import/no-unresolved
} from "!babel-loader!mdx-loader!../tos/acceptable-use-policy.mdx";
// eslint-disable-next-line import/no-webpack-loader-syntax
import PrivacyPolicy, {
  frontMatter as privacyFrontMatter,
  // eslint-disable-next-line import/no-unresolved
} from "!babel-loader!mdx-loader!../tos/privacy-policy.mdx";
// eslint-disable-next-line import/no-webpack-loader-syntax
import Terms, {
  frontMatter as termsFrontMatter,
  // eslint-disable-next-line import/no-unresolved
} from "!babel-loader!mdx-loader!../tos/terms-of-service.mdx";

import "../tos/tos-page.scss";

initFirebase();

const useStyles = makeStyles((theme) => ({
  root: {
    backgroundColor: theme.palette.background.paper,
  },
  tabPanel: {
    maxHeight: 300,
    [theme.breakpoints.up("sm")]: {
      maxHeight: 500,
    },
    overflow: "auto",
    padding: "1rem",
  },
  actions: {
    display: "flex",
    flexWrap: "wrap",
    alignItems: "center",
    justifyContent: "flex-end",
    padding: "0.75rem",
  },
}));

export const useLoadingUserTos = (
  loadingUser,
  user,
  setUserTos,
  setError,
  setLoading
) => {
  useEffect(() => {
    const loadUserTos = async () => {
      if (!loadingUser) {
        try {
          const snap = await firebase
            .database()
            .ref(`/users/${user.wbid}/tos`)
            .once("value");

          if (snap.exists()) {
            setUserTos(snap.val());
          } else {
            setUserTos({});
          }
        } catch (e) {
          setError(e.message);
        } finally {
          setLoading(false);
        }
      }
    };
    loadUserTos();
  }, [loadingUser, user.wbid, setUserTos, setError, setLoading]);
};

export const TosCheck = (props) => {
  const { loading = null, children = null } = props;

  const { user, loadingUser } = useUser();
  const [isLoading, setLoading] = React.useState(true);
  const [error, setError] = React.useState();
  const [userTos, setUserTos] = React.useState({});
  const muiTheme = useTheme();
  const classes = useStyles();
  const [tabValue, setTabValue] = React.useState(0);

  const [alerts, setAlerts] = React.useState({});
  const [working, setWorking] = React.useState({});

  /**
   * Add an alert to the dialog
   *
   * Parameters:
   *
   * id: A unique ID
   * alertProps: Bootstrap Alert props. See https://react-bootstrap.github.io/components/alerts/
   * content: The alert content
   */
  const addAlert = (id, alertProps, content) => {
    setAlerts({
      ...alerts,
      [id]: {
        props: alertProps,
        content,
      },
    });
  };

  const dismissAlert = (id) => {
    if (alerts[id]) {
      delete alerts[id];
      const newAlerts = {
        ...alerts,
      };
      setAlerts(newAlerts);
    }
  };

  const addWorking = (id, isWorking) => {
    setWorking({
      ...working,
      [id]: isWorking,
    });
  };

  const handleChange = (event, newValue) => {
    setTabValue(newValue);
  };

  const handleChangeIndex = (index) => {
    setTabValue(index);
  };

  useLoadingUserTos(loadingUser, user, setUserTos, setError, setLoading);

  const wrapContent = (content) => {
    const renderedAlerts = [];

    _forEach(alerts, (alert, alertId) => {
      renderedAlerts.push(
        <Alert
          {...alert.props}
          key={alertId}
          dismissible
          onClose={() => {
            dismissAlert(alertId);
          }}
        >
          {alert.content}
        </Alert>
      );
    });

    return (
      <>
        <SimpleNavBar />
        <div className="container-fluid tos-page">
          <div>
            <br />
            <br />
            <br />
            <div className="margin-middle tos-body">
              {renderedAlerts}
              {content}
            </div>
          </div>
        </div>
      </>
    );
  };

  if (isLoading || loadingUser) {
    return loading;
  }

  if (error) {
    return wrapContent(<Alert variant="danger">{error}</Alert>);
  }

  const latestTos = {
    terms: {
      ...termsFrontMatter,
      type: "terms",
      component: Terms,
      sortOrder: 0,
    },
    acceptableUse: {
      ...aupFrontMatter,
      type: "acceptableUse",
      component: AcceptableUsePolicy,
      sortOrder: 1,
    },
    privacy: {
      ...privacyFrontMatter,
      type: "privacy",
      component: PrivacyPolicy,
      sortOrder: 2,
    },
  };

  const needsAccept = [];

  _forEach(latestTos, (tos, tosType) => {
    if (tos.lastUpdated !== userTos[tosType]) {
      needsAccept.push(tos);
    }
  });

  if (needsAccept.length) {
    needsAccept.sort((a, b) => {
      return a.sortOrder - b.sortOrder;
    });

    const firstTime = !Object.keys(userTos).length;
    let title = "";
    if (firstTime) {
      title =
        "In order to proceed, you must accept the following terms of service:";
    } else {
      title =
        "Our terms of service have been updated. In order to proceed, you must accept the following:";
    }

    const handleSubmit = async (event) => {
      event.preventDefault();

      const workId = "tos-agree-proceed";

      addWorking(workId, true);

      try {
        const updates = {};
        needsAccept.forEach((tos) => {
          updates[tos.type] = tos.lastUpdated;
        });
        await firebase
          .database()
          .ref(`/users/${user.wbid}/tos`)
          .update(updates);
        setUserTos({
          ...userTos,
          ...updates,
        });
      } catch (e) {
        addAlert(
          workId,
          {
            variant: "danger",
          },
          e.message
        );
      } finally {
        addWorking(workId, false);
      }
    };

    const content = (
      <form className={classes.root} onSubmit={handleSubmit}>
        <h1 className="h3">{title}</h1>
        <br />
        <TabContext value={String(tabValue)}>
          <AppBar position="static" color="default">
            <TabList
              value={tabValue}
              onChange={handleChange}
              indicatorColor="primary"
              textColor="primary"
              variant="fullWidth"
            >
              {needsAccept.map((tos, i) => {
                return (
                  <Tab key={tos.type} value={String(i)} label={tos.tabLabel} />
                );
              })}
            </TabList>
          </AppBar>
          <SwipeableViews
            axis={muiTheme.direction === "rtl" ? "x-reverse" : "x"}
            index={Number(tabValue)}
            onChangeIndex={handleChangeIndex}
          >
            {needsAccept.map((tos, i) => {
              return (
                <TabPanel
                  key={tos.type}
                  value={String(i)}
                  dir={muiTheme.direction}
                  className={classes.tabPanel}
                >
                  <tos.component />
                </TabPanel>
              );
            })}
          </SwipeableViews>
          <div className={classes.actions}>
            <button
              type="submit"
              className="confirm-button"
              disabled={working["tos-agree-proceed"]}
            >
              Agree
            </button>
          </div>
        </TabContext>
      </form>
    );

    return wrapContent(content);
  }

  return children;
};

TosCheck.propTypes = {
  loading: PropTypes.node,
  children: PropTypes.node,
};

export default TosCheck;
