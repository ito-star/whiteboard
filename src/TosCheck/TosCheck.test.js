import React from "react";
import { shallow } from "enzyme";
import { renderHook, act } from "@testing-library/react-hooks";
import firebase from "firebase/compat/app";
import "firebase/compat/database";
import useUser from "../auth/useUser";
import { TosCheck, useLoadingUserTos } from "./TosCheck";
import Loader from "../components/Loader";

import AccountPageToMock from "./__mocks__/AccountPageToMock";

jest.mock("../auth/useUser", () => jest.fn());

jest.mock("babel-loader!mdx-loader!../tos/terms-of-service.mdx", () => ({
  frontMatter: { lastUpdated: "tpday" },
}));
jest.mock("babel-loader!mdx-loader!../tos/acceptable-use-policy.mdx", () => ({
  frontMatter: { lastUpdated: "tpday" },
}));
jest.mock("babel-loader!mdx-loader!../tos/privacy-policy.mdx", () => ({
  frontMatter: { lastUpdated: "tpday" },
}));
jest.mock("../utils", () => ({
  initFirebase: jest.fn(),
  scrollToRefObject: jest.fn(),
  mbToBytes: jest.fn(),
  createUUID: jest.fn(),
}));

// jest.mock("firebase");
jest.mock("firebase/compat/app");
jest.mock("firebase/compat/database");

describe("TosCheck", () => {
  const userData = {
    user: {
      wbid: "johndoes@gmail.com",
      uid: "W211SxVXAEhvmWBmn0nwi1WI8wT2",
      isAnonymous: false,
      email: "johndoes@gmail.com",
      emailVerified: true,
      displayName: "John Doe",
      photoURL: "https://lh3.googleusercontent.com/-0xxalooz21s/AAA",
      token: { claims: { isSpecial: false } },
      usage: { boards: 0 },
    },
  };
  const mockFn = jest.fn();

  afterEach(() => {
    jest.clearAllMocks();
  });

  beforeEach(() => {
    useUser.mockImplementation(() => userData);
  });

  const defaultProps = {
    loading: <Loader isFullScreen />,
    children: [<AccountPageToMock />],
  };

  describe("with props", () => {
    it("render default", () => {
      const component = shallow(<TosCheck {...defaultProps} />);

      expect(component).toMatchSnapshot();
    });

    it("render error", () => {
      const setLoading = mockFn;
      const setError = mockFn;
      const setAlerts = mockFn;
      const setUserTos = mockFn;
      const setTabValue = mockFn;
      const alerts = [{ props: {}, content: "Error Displayed", alertId: "34" }];
      const working = {};
      const setWorking = mockFn;

      jest
        .spyOn(React, "useState")
        .mockImplementationOnce(() => [false, setLoading])
        .mockImplementationOnce(() => [true, setError])
        .mockImplementationOnce(() => [{}, setUserTos])
        .mockImplementationOnce(() => ["1", setTabValue])
        .mockImplementationOnce(() => [alerts, setAlerts])
        .mockImplementationOnce(() => [working, setWorking]);

      const component = shallow(<TosCheck {...defaultProps} />);

      expect(component.find("Alert").get(0).type.displayName).toBe("Alert");
      expect(component.find("Alert").get(0).type.Link.displayName).toBe(
        "AlertLink"
      );
      expect(component.find("Alert").get(0).type.Heading.displayName).toBe(
        "AlertHeading"
      );
      expect(component.find("Alert").get(0).type.defaultProps.show).toBe(true);
      expect(
        component.find("Alert").get(0).type.defaultProps.transition.defaultProps
      ).toStrictEqual({
        in: false,
        timeout: 300,
        mountOnEnter: false,
        unmountOnExit: false,
        appear: false,
      });
      expect(
        component.find("Alert").get(0).type.defaultProps.transition.displayName
      ).toBe("Fade");
      expect(component.find("Alert").get(0).type.defaultProps.closeLabel).toBe(
        "Close alert"
      );
      expect(component.find("Alert").get(0).props.dismissible).toBe(true);
      expect(component.find("Alert").get(0).props.show).toBe(true);
      expect(component.find("Alert").get(0).props.children).toBe(
        "Error Displayed"
      );
      expect(component.find("Alert").get(0).props.closeLabel).toBe(
        "Close alert"
      );
      component.find("Alert").get(0).props.onClose();

      expect(component.find("Alert").get(1).props.variant).toBe("danger");
      expect(component.find("Alert").get(1).props.children).toBe(true);
      expect(component.find("Alert").get(1).props.show).toBe(true);
      expect(component.find("Alert").get(1).props.transition.displayName).toBe(
        "Fade"
      );
      expect(component.find("Alert").get(1).props.variant).toBe("danger");
      expect(component.find("Alert").get(1).props.closeLabel).toBe(
        "Close alert"
      );
    });

    it("render firstTime message", () => {
      const setLoading = mockFn;
      const setError = mockFn;
      const setUserTos = mockFn;
      const setTabValue = mockFn;
      const setAlerts = mockFn;
      const alerts = {};
      const working = {};
      const setWorking = mockFn;
      jest
        .spyOn(React, "useState")
        .mockImplementationOnce(() => [false, setLoading])
        .mockImplementationOnce(() => [false, setError])
        .mockImplementationOnce(() => [{}, setUserTos])
        .mockImplementationOnce(() => ["1", setTabValue])
        .mockImplementationOnce(() => [alerts, setAlerts])
        .mockImplementationOnce(() => [working, setWorking]);

      const component = shallow(<TosCheck {...defaultProps} />);

      expect(component.find("h1").get(0).props).toStrictEqual({
        className: "h3",
        children:
          "In order to proceed, you must accept the following terms of service:",
      });
    });

    it("render terms of service updated message", () => {
      const setLoading = mockFn;
      const setError = mockFn;
      const setUserTos = mockFn;
      const setTabValue = mockFn;
      const setAlerts = mockFn;
      const alerts = {};
      const working = {};
      const setWorking = mockFn;
      jest
        .spyOn(React, "useState")
        .mockImplementationOnce(() => [false, setLoading])
        .mockImplementationOnce(() => [false, setError])
        .mockImplementationOnce(() => [
          { terms: "hello", tosType: "xyz" },
          setUserTos,
        ])
        .mockImplementationOnce(() => ["1", setTabValue])
        .mockImplementationOnce(() => [alerts, setAlerts])
        .mockImplementationOnce(() => [working, setWorking]);

      const component = shallow(<TosCheck {...defaultProps} />);

      expect(component.find("form").props().className).toBe(
        "makeStyles-root-1"
      );
      expect(component.find("h1").get(0).props).toStrictEqual({
        className: "h3",
        children:
          "Our terms of service have been updated. In order to proceed, you must accept the following:",
      });
      expect(component.find("TabContext").props().value).toBe("1");
      expect(
        component.find("WithStyles(ForwardRef(AppBar))").props().position
      ).toBe("static");
      expect(
        component.find("WithStyles(ForwardRef(AppBar))").props().color
      ).toBe("default");
      expect(component.find("ForwardRef(TabList)").props().value).toBe("1");
      component.find("ForwardRef(TabList)").props().onChange();
      expect(component.find("ForwardRef(TabList)").props().indicatorColor).toBe(
        "primary"
      );
      expect(component.find("ForwardRef(TabList)").props().textColor).toBe(
        "primary"
      );
      expect(component.find("ForwardRef(TabList)").props().variant).toBe(
        "fullWidth"
      );
      expect(
        component.find("WithStyles(ForwardRef(Tab))").length
      ).toStrictEqual(3);
      expect(component.find("ReactSwipableView")).toBeTruthy();
      component.find("ReactSwipableView").props().onChangeIndex();
      expect(component.find("WithStyles(ForwardRef(TabPanel))")).toBeTruthy();
      expect(component.find("terms-of-service.mdx")).toBeTruthy();
      expect(component.find("button").props()).toStrictEqual({
        type: "submit",
        className: "confirm-button",
        disabled: undefined,
        children: "Agree",
      });
    });

    it("render children", () => {
      const setLoading = mockFn;
      const setError = mockFn;
      const setUserTos = mockFn;
      const setTabValue = mockFn;
      const setAlerts = mockFn;
      const alerts = {};
      const working = {};
      const setWorking = mockFn;
      jest
        .spyOn(React, "useState")
        .mockImplementationOnce(() => [false, setLoading])
        .mockImplementationOnce(() => [false, setError])
        .mockImplementationOnce(() => [
          { terms: "tpday", acceptableUse: "tpday", privacy: "tpday" },
          setUserTos,
        ])
        .mockImplementationOnce(() => ["1", setTabValue])
        .mockImplementationOnce(() => [alerts, setAlerts])
        .mockImplementationOnce(() => [working, setWorking]);

      const component = shallow(<TosCheck {...defaultProps} />);

      expect(component.find("AccountPageToMock")).toBeTruthy();
    });
  });

  describe("Use effect", () => {
    afterEach(() => {
      jest.clearAllMocks();
    });

    /**
     * @todo FIX ME
     *
     * I think something about the refactoring of the Firebase
     * imports messed with the mocks. Unfortunately, I do not
     * know enough about Jest to fix this. -- Aaron
     */
    it.skip("setUserTos when snap exists", async () => {
      const setUserTos = jest.fn();
      const setError = jest.fn();
      const setLoading = jest.fn();

      const val = { user: "mockValue" };
      const snap = {
        exists: () => true,
        val: () => val,
      };

      const once = jest.fn(async () => snap);
      const ref = jest.fn(() => ({ once }));
      firebase.database.mockImplementation(() => ({ ref }));

      await act(async () => {
        renderHook(() => {
          useLoadingUserTos(
            false,
            userData.user,
            setUserTos,
            setError,
            setLoading
          );
        });
      });

      expect(setUserTos).toHaveBeenCalledTimes(1);
      expect(setUserTos).toHaveBeenCalledWith(val);
      expect(setLoading).toHaveBeenCalledTimes(1);
      expect(setLoading).toHaveBeenCalledWith(false);
    });

    /**
     * @todo FIX ME
     *
     * I think something about the refactoring of the Firebase
     * imports messed with the mocks. Unfortunately, I do not
     * know enough about Jest to fix this. -- Aaron
     */
    it.skip("setUserTos when snap does not exists", async () => {
      const setUserTos = jest.fn();
      const setError = jest.fn();
      const setLoading = jest.fn();

      const val = {};
      const snap = {
        exists: () => false,
        val: () => val,
      };

      const once = jest.fn(async () => snap);
      const ref = jest.fn(() => ({ once }));
      firebase.database.mockImplementation(() => ({ ref }));

      await act(async () => {
        renderHook(() => {
          useLoadingUserTos(
            false,
            userData.user,
            setUserTos,
            setError,
            setLoading
          );
        });
      });

      expect(setUserTos).toHaveBeenCalledTimes(1);
      expect(setUserTos).toHaveBeenCalledWith(val);
      expect(setLoading).toHaveBeenCalledTimes(1);
      expect(setLoading).toHaveBeenCalledWith(false);
    });

    /**
     * @todo FIX ME
     *
     * I think something about the refactoring of the Firebase
     * imports messed with the mocks. Unfortunately, I do not
     * know enough about Jest to fix this. -- Aaron
     */
    it.skip("setUserTos when snap does not exists", async () => {
      const setUserTos = jest.fn();
      const setError = jest.fn();
      const setLoading = jest.fn();

      firebase.database.mockImplementation(() => {
        throw Error("error");
      });

      await act(async () => {
        renderHook(() => {
          useLoadingUserTos(
            false,
            userData.user,
            setUserTos,
            setError,
            setLoading
          );
        });
      });

      expect(setError).toHaveBeenCalledTimes(1);
      expect(setError).toHaveBeenCalledWith("error");
      expect(setLoading).toHaveBeenCalledTimes(1);
      expect(setLoading).toHaveBeenCalledWith(false);
    });

    it("loading user is true", async () => {
      const setUserTos = jest.fn();
      const setError = jest.fn();
      const setLoading = jest.fn();

      await act(async () => {
        renderHook(() => {
          useLoadingUserTos(
            true,
            userData.user,
            setUserTos,
            setError,
            setLoading
          );
        });
      });

      expect(setLoading).toHaveBeenCalledTimes(0);
    });
  });
});
