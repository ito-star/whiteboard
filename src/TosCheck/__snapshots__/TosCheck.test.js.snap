// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TosCheck with props render default 1`] = `ShallowWrapper {}`;
